private/bufpkg/bufconnect/interceptors.go

```diff
@@ -42,65 +43,136 @@ func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
 	return interceptor
 }
 
+// TokenFinder finds the token for NewAuthorizationInterceptorProvider
+type TokenFinder interface {
+	// RemoteToken returns the remote token from the remote address.
+	// setFromEnvVar is true if the returned token is from the tokenEnvKey environment variable.
```

Comment:
Small doc nit: that's a private name and a detail. (and ugly detail, but one anyway)

```suggestion
	// RemoteToken returns the remote token from the remote address.
	// setFromEnvVar is true if the returned token is from an environment variable.
```